src/cargo/core/workspace.rs

```diff
@@ -873,11 +873,7 @@ impl<'cfg> Workspace<'cfg> {
                 MaybePackage::Package(ref p) => p.clone(),
                 MaybePackage::Virtual(_) => continue,
             };
-            let mut src = PathSource::new(
-                pkg.manifest_path(),
-                pkg.package_id().source_id(),
-                self.config,
-            );
+            let mut src = PathSource::new(pkg.root(), pkg.package_id().source_id(), self.config);
```

Comment:
I think there's one more case of this going wrong. [vendor](https://github.com/rust-lang/cargo/blob/8aa332deaadf1a54646d8d7f9a4ac6dc422b2218/src/cargo/ops/vendor.rs#L216) also passes the manifest path. I think a cursory looks suggests that's the last case though.

Reply:
Ah I think `.parent()` is called there so the directory is passed in (I tried to audit all sources of `PathSource::new` to make sure the directory, not the manifest, was passed in)

Reply:
Ah yes, I missed that -- seems like it should be calling `Package::root()` rather than manually getting the parent though. But indeed not an issue.